src/components/NewCommentForm.tsx

```diff
@@ -57,15 +60,14 @@ export const NewCommentForm: React.FC<Props> = ({ onSubmit }) => {
       return;
     }
 
-    setSubmitting(true);
-
-    // it is very easy to forget about `await` keyword
-    await onSubmit({ name, email, body });
-
-    // and the spinner will disappear immediately
-    setSubmitting(false);
+    dispatch(addComment({
+      id: 0,
```

Comment:
Instead of assigning id as 0 hero you can change the type in commentSlice for the addComment function so that you wouldn't need to add id here.